face/VatomView.js

```diff
@@ -0,0 +1,119 @@
+    }
+
+    this.update();
+  }
+
+  update() {
+    // Check if ready to be displayed
+    if (!this.vatomObj) {
+      console.warn('No vAtom supplied');
+      // eslint-disable-next-line
+      return;
+      // eslint-disable-next-line
+    } else {
+      this.free();
+      this.load();
+    }
+  }
+
+  load() {
+    // start the face selection procedure
+    const st = this.fsp(this.vatomObj.faces);
+    // check if face is registered
+    const du = st.properties.display_url;
+    let FaceClass = registeredFace[du];
+    // if there is no face registered in the array but we have a http link, show the web face
+    if (FaceClass === undefined && du.indexOf('http') !== -1) {
+      FaceClass = ImageFace;
+    } else if (FaceClass === undefined && du.indexOf('http') === -1) {
+      // display activated Image in the case that the face is undefined!
+      FaceClass = ImageFace;
+    }
+    // create a new instance of the chosen face class and pass through the information
+    const rFace = new FaceClass(this, this.vatomObj, st);
+    this._currentFace = rFace;
+    // make rface opaque
+    rFace.element.style.opacity = 0;
+    // add face to element
+    this.element.appendChild(rFace.element);
+
+    // add the loader
+    this.element.appendChild(this.loader);
+    // call rface.onload , wait for promise
+    Promise.resolve(() => null).then(() => rFace.onLoad()).then(() => {
+      this.element.removeChild(this.loader);
+      rFace.element.style.opacity = 1;
+    });
+  }
+
+  set vatom(vAtom) {
+    if (vAtom) {
+      this.vatomObj = vAtom;
+      this.update();
+    }
+  }
+
+  get vatom() {
+    return this.vatomObj;
+  }
+
+  free() {
+    if (this._currentFace) {
+      this._currentFace.onUnload();
+    }
+
+    const view = this.element;
+    while (view.firstChild) {
+      view.removeChild(view.firstChild);
+    }
+  }
+
+  // register our own face
+  static registerFace(faceClass) {
+    registeredFace[faceClass.url] = faceClass;
```

Comment:
To make urls case-insensitive, you can just change it to `faceClass.url.toLowerCase()` here...